### PR TITLE
Implement update vendor thing ID

### PR DESCRIPTION
**Pull request by dairyo**
* Move `ThingIFAPI.update` method from ThingIFAPI.swift to ThingIFAPI+ThingInformation.swift
* Add small tests.
* Delete an old test file.

Related PR: #296

### CODE REVIEW

File: ThingIFSDK/ThingIFSDK/IoTOperations/IoTRequestOperation.swift [hunk 1]
```diff
@@ -13,7 +13,9 @@ internal enum HTTPMethod: String {
     case GET = "GET-deprecated"
     case get = "GET"
     case post = "POST"
-    case PUT = "PUT"
+    @available(iOS, deprecated: 1.0, message: "use put")
+    case PUT = "PUT-deprecated"
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/IoTOperations/IoTRequestOperation.swift:17]: why not just removing `PUT` and other capital enum?

**Review by yongpingchen — CHANGES_REQUESTED** [review]

**Review by naoki-akimoto — APPROVED** [review]
LGTM!

File: ThingIFSDK/ThingIFSDK/IoTOperations/IoTRequestOperation.swift (hunk 1 above)
Comment (dairyo) [reply ThingIFSDK/ThingIFSDK/IoTOperations/IoTRequestOperation.swift:17]: @yongpingchen 

If we replace All `PUT` to `put` in this PR, we must make some unrelated fixes for this PR. I think it is not good PR which includes unrelated fixes for the purpose of the PR.

XCode points out the code using `PUT` with `@availale` annotation. I will replace them when needed.

Comment (yongpingchen) [reply ThingIFSDK/ThingIFSDK/IoTOperations/IoTRequestOperation.swift:17]: > XCode points out the code using PUT with @availale annotation.

I see. 

> If we replace All PUT to put in this PR, we must make some unrelated fixes for this PR. I think it is not good PR which includes unrelated fixes for the purpose of the PR.

I agree with this. 

My point is in 1.0, it is better to have clean version, without the deprecate annotations.

Comment (dairyo) [reply ThingIFSDK/ThingIFSDK/IoTOperations/IoTRequestOperation.swift:17]: @yongpingchen The version is iOS version. It is not our SDK version. We can use `PUT` in any versions of iOS so we can not clearly specify the version. Changing `PUT` to `put` comes from API guide line.

We must change all only uppercase enum element to camel case. My point for using `@availabe` is pointing out the code to change in XCode.

According to above opinion, version number is not important in my position.

What do you think?

Comment (yongpingchen) [reply ThingIFSDK/ThingIFSDK/IoTOperations/IoTRequestOperation.swift:17]: @shinsuke-nara let us keep this topic. I can merge this PR first

**Pull request merged**